Remotion/Data/DomainObjects.UnitTests/Persistence/Rdbms/RdbmsProviderTest.cs

```diff
@@ -70,11 +70,56 @@ public override void SetUp ()
 
       _provider = new TestableRdbmsProvider(
           TestDomainStorageProviderDefinition,
+          TestDomainStorageProviderDefinition.ConnectionString,
           NullPersistenceExtension.Instance,
           _commandFactoryMock.Object,
           () => _connectionCreatorMock.Object.CreateConnection());
     }
 
+
+    [Test]
+    public void RdbmsProvider_WithConnectionStringContainedInProviderDefinition ()
+    {
+      var rdbmsProvider = new RdbmsProvider(
+          new RdbmsProviderDefinition("dummyProviderDefinition", Mock.Of<IRdbmsStorageObjectFactory>(), "dummyConnectionString", "dummyReadOnlyConnectionString"),
+          "dummyConnectionString",
+          NullPersistenceExtension.Instance,
+          _commandFactoryMock.Object,
+          () => _connectionCreatorMock.Object.CreateConnection());
+
+      Assert.That(rdbmsProvider, Is.InstanceOf<RdbmsProvider>());
+    }
+
+    [Test]
+    public void RdbmsProvider_WithReadOnlyConnectionStringContainedInProviderDefinition ()
+    {
+      var rdbmsProvider = new RdbmsProvider(
+          new RdbmsProviderDefinition("dummyProviderDefinition", Mock.Of<IRdbmsStorageObjectFactory>(), "dummyConnectionString", "dummyReadOnlyConnectionString"),
+          "dummyReadOnlyConnectionString",
+          NullPersistenceExtension.Instance,
+          _commandFactoryMock.Object,
+          () => _connectionCreatorMock.Object.CreateConnection());
+
+      Assert.That(rdbmsProvider, Is.InstanceOf<RdbmsProvider>());
+    }
+
+    [Test]
+    public void RdbmsProvider_WithConnectionStringNotInProviderDefinition_ThrowsArgumentException ()
+    {
+      var rdbmsCtorCall = () => new RdbmsProvider(
+          new RdbmsProviderDefinition("dummyProviderDefinition", Mock.Of<IRdbmsStorageObjectFactory>(), "dummyConnectionString", "dummyReadOnlyConnectionString"),
+          "incorrectConnectionString",
+          NullPersistenceExtension.Instance,
+          _commandFactoryMock.Object,
+          () => _connectionCreatorMock.Object.CreateConnection());
+
+      Assert.That(
+          rdbmsCtorCall,
+          Throws.ArgumentException.With.ArgumentExceptionMessageEqualTo(
+              "The connection string 'incorrectConnectionString' is not defined by provider 'RdbmsProviderDefinition: 'dummyProviderDefinition''",
```

Comment:
there's two single-quotes at the end. plz fix.

Reply:
I'm not sure that would be correct:
We currently have `'bla: 'bla2''`, I don't think we want `'bla: 'bla2'`, or am I wrong here?

Reply:
@Frazerus ah, I see, and now I also understand why I didn't notice it in the production code first.

The fact that there's two single quotes is correct. And it's also an indicator that we're not building a message as well as we should (this kind of accidental string joining is in indicator of a design flaw in the message) and need to how to improve this.

The problem occurs because the message is passed the `RdmbsProviderDefintion` as an object instead of `RdbmsProviderDefintion.Name`. For the most part, a ToString-version of a complex object isn't what we want in an argument etc exception. They're mostly for debugger and diagnostics.